hearpreprocess/pipeline.py

```diff
@@ -533,7 +533,7 @@ def trim_event_metadata(self, metadata: pd.DataFrame, duration: float):
 
         # Trim the events starting before but extending beyond the sample duration
         events_trimmed = len(trimmed_metadata.loc[lambda df: df["end"] > duration_ms])
-        trimmed_metadata.loc[lambda df: df["end"] > duration_ms] = duration_ms
+        trimmed_metadata.loc[lambda df: df["end"] > duration_ms, "end"] = duration_ms
```

Comment:
This seems like a really weird place to do this. Because I thought ExtractMetadata has a contract (maybe undocumented?) that the metadata is the original labels, not altered.

Can this go somewhere else that is more sensible? Maybe even its own task? Or in the Trim task?

Or if it does make sense that this goes here, can you explain why in a longer comment?

trim_event_metadata needs a docstring.

Why do we only

Reply:
Why above do we do:

```
trimmed_metadata = metadata.loc[lambda df: df["start"] < duration_ms]
```

instead of

```
trimmed_metadata = metadata[metadata["start"] < duration_ms]
```

Also, should this be <= ?

Reply:
In the current pipeline setup, we use the same metadata throughout the pipeline and this comes from `ExtractMetadata`, so any change to metadata has to be done here as it is not manipulated anywhere. All the other tasks handle the audio files and manipulate them, but the split of metadata, generating the metadata is done in the `ExtractMetadata`. So added this in the `ExtractMetadata`.

Reply:
`trimmed_metadata = metadata.loc[lambda df: df["start"] < duration_ms]`
can well be replaced by 
`trimmed_metadata = metadata[metadata["start"] < duration_ms]`
I follow the lambda function as a habit because when we pipe operators, the previous function might have changed the dataframe and lambda helps in that case, will change here to be straight forward. 

For the `<` to `<=`, I have kept `<` for start and `<=` for the end. Basically, to be counted as an event, anything should start before the end of the file. However, any ongoing event can go till the end. Should we change this behaviour?

Reply:
"
For the < to <=, I have kept < for start and <= for the end. Basically, to be counted as an event, anything should start before the end of the file. However, any ongoing event can go till the end. Should we change this behaviour?" Sounds good document this

Reply:
" the previous function might have changed the dataframe and lambda helps in that case" Can you explain? I'm just also new to pandas

Reply:
Its like 
In `metadata.someoperation.loc[lambda df: df["start"] < duration_ms]`. the loc will be applied on `metadata.someoperation`, 
but if we do `metadata.someoperation.loc[metadata["start"] < duration_ms]`, the command will fail and we will have to do `metadata.someoperation.loc[metadata.someoperation["start"] < duration_ms]` to make it work.
So, when we chain lots of command we should do `lambda df: `, it picks up the dataframe till the previous operation.

Example of `someoperation` can be any chained pandas operation.
But, in this case, since there is no chained event, we can use the simple `metadata[metadata["start"] < duration_ms]`